CRM/Contribute/BAO/Contribution.php

```diff
@@ -4464,8 +4465,8 @@ public static function completeOrder($input, &$ids, $objects, $transaction = NUL
     ));
 
     // CRM-20678 Ensure that the currency is correct in subseqent transcations.
-    if (empty($contributionParams['currency']) && isset($objects['first_contribution']->currency)) {
-      $contributionParams['currency'] = $objects['first_contribution']->currency;
+    if (empty($contributionParams['currency']) && isset($contribution->currency)) {
```

Comment:
We need a test covering this -and it definitely needs moving to within repeattransaction function as it only applies for repeats. It would be good to tackle & resolve this first

Reply:
`$contribution` object is `$objects['contribution']` which is a copy of `$objects['first_contribution']` with 3 parameters removed (id, receive_date and invoice_id). So currency at this point is the same value.
Agree with moving, not sure it needs it's own test because I'm happy from code analysis that it's the same value. Would prefer to move after this PR

Reply:
I'm pretty sure these lines can go now - we should just add currency to that test  - that might be quick to do - looking

Reply:
OK I confirmed they can go https://github.com/civicrm/civicrm-core/pull/18055